openmc/mgxs/library.py

```diff
@@ -773,9 +773,9 @@ def get_xsdata(self, domain, xsdata_name, nuclide='total', xs_type='macro',
             nuclide this will be set to 'macro' regardless.
         xs_ids : str
             Cross section set identifier. Defaults to '1m'.
-        order : Scattering order for this dataset entry.  Default is None,
-            which will force the XSdata object to use whatever the maximum
-            order available.
+        order : Scattering order for this data entry.  Default is None,
+            which will force the XSdata object to use whatever the order of the
```

Comment:
I originally chose `order` when writing this (before #644 was submitted) because it matches the nomenclature within the OpenMC MGXS input format.  Now, the question then is 'why does the OpenMC MGXS input format call it `order`? The reason is because the scattering anisotropy can be represented by Legendre moments, mu histograms, and/or mu tabular distributions.  If Legendres, order means Legendre expansion order, if histograms, the histogram order is a synonym for the number of bins, if tabular, the tabular order is a synonym for the number of points.  

When updating this for your new code last night, I use your `legendre_order` as is since right now the Mgxs and Library classes only support the Legendre expansion, but it doesn't always have to, so I chose to keep this terminology for future's sake.